integration_tests/__tests__/clear_cache.test.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+'use strict';
+
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+const {cleanup, writeFiles} = require('../utils');
+const runJest = require('../runJest');
+const skipOnWindows = require('../../scripts/skip_on_windows');
+
+const CACHE = path.resolve(os.tmpdir(), 'clear_cache_directory');
+const DIR = path.resolve(os.tmpdir(), 'clear_cache');
+
+skipOnWindows.suite();
+
+beforeEach(() => cleanup(DIR));
+afterAll(() => cleanup(DIR));
+
+describe('jest --clearCache', () => {
+  test('clearing cache results in exit status 0', () => {
+    writeFiles(DIR, {
+      '.watchmanconfig': '',
+      'package.json': '{}',
+    });
+
+    const {status, stdout, stderr} = runJest(DIR, [
+      '--clearCache',
+      `--cacheDirectory=${CACHE}`,
+    ]);
+
+    expect(fs.existsSync(CACHE)).toBe(false);
```

Comment:
It's a false positive. This folder doesn't exist, because Jest didn't run any tests. We need to populate this cache earlier with a sample run, then run it once more. Or just create the dir and make sure it is present prior to running test.

Reply:
Sorry I thought the folder was created when calling runJest regardless of whether there are tests or not. Will add a stub test, verify the cache dir has contents, then runJest again with clearCache

Reply:
can you add

```js
expect(fs.existsSync(CACHE)).toBe(true);
```

above the `runJest` call?

Reply:
In the second test correct? Makes sense.

Reply:
Correct!